core/base/inc/TROOT.h

```diff
@@ -80,12 +80,21 @@ namespace Internal {
 } } // End ROOT::Internal
 
 namespace ROOT {
+   enum class EIMTConfig {
+      kWholeMachine = 0,     ///< Default configuration
+      kExistingTBBArena = 1, ///< Use the existing TBB arena
+      kNumConfigs = 2        ///< Number of support IMT semantic configurations
+
+   };
    /// \brief Enable support for multi-threading within the ROOT code
    /// in particular, enables the global mutex to make ROOT thread safe/aware.
    void EnableThreadSafety();
    /// \brief Enable ROOT's implicit multi-threading for all objects and methods that provide an internal
    /// parallelisation mechanism.
    void EnableImplicitMT(UInt_t numthreads = 0);
+   /// \brief Enable ROOT's implicit multi-threading for all objects and methods that provide an internal
+   /// parallelisation mechanism.
+   void EnableImplicitMT(ROOT::EIMTConfig config);
```

Comment:
Would a new function instead of an overload make sense here? In terms of documenting (or looking up) the meaning of the arguments, especially in doxygen, wouldn't it be easier to have a second function:
```c++
EnableImplicitMT(unsigned int numthreads);
EnableImplicitMTExistingTaskArena();
```

Reply:
> even `kMPI`

That doesn't make much sense because it's not multi*threading*

Reply:
Still, if we are going to add a new interface, i'll rather add a flexible one.